contract_reader.py

```diff
@@ -115,11 +104,62 @@ def get_orderbook():
             "Read order: sellAmount %40d %7s -- buyAmount %40d %7s -- accountID %s"
             % (sell_amount, sell_token, buy_amount, buy_token, o['accountID']))
 
-        orders.append({'accountID': o['accountID'],
-                       'sellToken': sell_token,
-                       'buyToken': buy_token,
-                       'sellAmount': str(int(sell_amount)),
-                       'buyAmount': str(int(buy_amount))})
+        orders_decoded.append({'accountID': o['accountID'],
+                               'sellToken': sell_token,
+                               'buyToken': buy_token,
+                               'sellAmount': str(int(sell_amount)),
+                               'buyAmount': str(int(buy_amount)),
+                               'validFrom': o['validFrom'],
+                               'validUntil': o['validUntil']})
+
+    return orders_decoded
+
+
+def get_current_orderbook():
+    """Get all order data from smart contract.
+
+    Returns:
+        A list of all orders.
+
+    """
+    orders = get_orderbook()
+
+    # Skip orders that are no longer or not yet valid.
+    batch_id = get_current_batch_id()
+    orders = list(filter(lambda order: order['validUntil'] <
+                         batch_id or order['validFrom'] > batch_id, orders))
```

Comment:
1. Not sure I understand what we want here, but If `orders` is the set of orders that include the current batch_id, then the condition should be `validUntil>batchId and validFrom<batchId. 

2. I never know what the convention is, but I find it difficult to believe that this interval is open on both sides. It should be closed on one of them, right?

Reply:
I just copy-pasted the line from the current python script. But in there,  it used the inverse logic. Thanks for highlighting!